### PR TITLE
Generate unique activity task token per attempt

### LINKED ISSUE #84: TestWorkflowService: Make tokens unique across retries
Currently real service doesn't accepts activity and workflow task tokens across retries. But the TestWorkflowService [includes only workflow and activity id in it](https://github.com/temporalio/temporal-java-sdk/blob/538557e55d27635bd08e507bc4b595c19e20223a/src/main/java/io/temporal/internal/testservice/TestWorkflowService.java#L544).

### LINKED ISSUE #1223: Test server generates the same activity task token on retry
## Expected Behavior

Should be a different task token

## Actual Behavior

Same token issued

## Steps to Reproduce the Problem

  1. Create a test with an activity that times out
  1. Wait for second attempt
  1. Observe the issued task token

### PR DESCRIPTION
**Pull request by bergundy**
Closes #84
Didn't add tests yet, making sure this is the right direction due to lack of familiarity with the codebase.

### CODE REVIEW

File: temporal-test-server/src/main/java/io/temporal/internal/testservice/ActivityIdWithAttempt.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
+package io.temporal.internal.testservice;
+
+import com.google.protobuf.ByteString;
+import io.grpc.Status;
+import io.temporal.api.common.v1.WorkflowExecution;
+
+import java.io.*;
+
+public class ActivityIdWithAttempt {
```
Comment (Spikhalskiy) [inline temporal-test-server/src/main/java/io/temporal/internal/testservice/ActivityIdWithAttempt.java:9]: Doesn't need to be public. Just remove it, making it package-private.

File: temporal-test-server/src/main/java/io/temporal/internal/testservice/ActivityId.java [hunk 2]
```diff
@@ -89,17 +89,21 @@ public ByteString toBytes() {
...
-      out.writeUTF(execution.getRunId());
-      out.writeLong(scheduledEventId);
+      this.writeBytes(out);
       return ByteString.copyFrom(bout.toByteArray());
     } catch (IOException e) {
       throw Status.INTERNAL.withCause(e).withDescription(e.getMessage()).asRuntimeException();
     }
   }
 
+  public void writeBytes(DataOutputStream out) throws IOException {
```
Comment (Spikhalskiy) [inline temporal-test-server/src/main/java/io/temporal/internal/testservice/ActivityId.java:99]: Doesn't need to be public. Just remove it, making it package-private.

File: temporal-test-server/src/main/java/io/temporal/internal/testservice/ActivityId.java [hunk 3]
```diff
@@ -108,16 +112,20 @@ static ActivityId fromBytes(byte[] serialized) {
...
-      String runId = in.readUTF();
-      long scheduledEventId = in.readLong();
-      return new ActivityId(namespace, workflowId, runId, scheduledEventId);
+      return ActivityId.readBytes(in);
     } catch (IOException e) {
       throw Status.INTERNAL.withCause(e).withDescription(e.getMessage()).asRuntimeException();
     }
   }
 
+  public static ActivityId readBytes(DataInputStream in) throws IOException {
```
Comment (Spikhalskiy) [inline temporal-test-server/src/main/java/io/temporal/internal/testservice/ActivityId.java:121]: It's `fromBytes` in `ActivityIdWithAttempt`, let's make `fromBytes` here too

**Review by Spikhalskiy — APPROVED** [review]

**Review by Spikhalskiy — APPROVED** [review]
Just some small comments.
Don't worry about test if you don't have time for it now, I can submit a test in the upcoming PR.
It looks like tests hang now though.

**Comment by bergundy** [comment]
I'll probably only get back to this tomorrow

**Comment by Spikhalskiy** [comment]
Closed due to inactivity in favor of #1524

**Pull request closed without merging**